mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp

```diff
@@ -618,7 +549,8 @@ LogicalResult PopulateParamsXDL::obtainTuningParameters(
 }
 
 std::vector<InitParamsXDL>
-PopulateParamsXDL::getTuningParameters(ConvOpType dir, Type dataType) const {
+PopulateParamsXDL::getTuningParameters(Optional<ConvOpType> dir,
```

Comment:
I think we can drop dir -- it is not being used.

Reply:
I left it in there because we might eventually have cases where backward data etc. have different sets of good heuristic parameters - we had those before, IIRC.

Reply:
How about we pass the Op directly instead as we are expanding this pass to rock.gemm now ?

Reply:
I'd love to do that but `Conv2DGenerator` calls this for the backward weight padding kernel stuff

Reply:
Hmmm.. I thought getTuningParameters is only called under affix params pass.

Just to clarify, my suggestion was to replace the dir with the Op (that already captures each convop corresponding to the direction as well as gemm op) here, assuming the clients of this function within affix params pass.

IIRC, there is a different function to obtain kernel counts, workspace, etc that also scrolls these things but it was not getTuningParameters

Reply:
The call is in `calculatePaddingKernelSize()` in `Conv2dGenerator::needExtraPadBwdWeight`

If it weren't for that one call, I would've made this take the op just like the rest of the system

Reply:
ah yes... I remember now.